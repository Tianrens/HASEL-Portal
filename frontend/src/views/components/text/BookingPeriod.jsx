import dayjs from 'dayjs';
import { React } from 'react';

const BookingPeriod = ({ startTimestamp, endTimestamp }) => {
    const startDateFormatted = dayjs(startTimestamp).format('ddd DD/MM/YY hh:mm A');
    const endDateFormatted = dayjs(endTimestamp).format('ddd DD/MM/YY hh:mm A');
    return (
        <span>
            {`${startDateFormatted}  -  ${endDateFormatted}`}
        </span>
    );
};

export default BookingPeriod;
