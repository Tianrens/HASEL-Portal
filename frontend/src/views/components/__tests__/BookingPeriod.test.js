import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider } from '@material-ui/core';
import BookingPeriod from '../text/BookingPeriod';

test('BookingPeriod renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <BookingPeriod startTimestamp='2021-10-06T05:46:00.000+00:00' endTimestamp='2021-10-10T05:46:00.000+00:00' />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
