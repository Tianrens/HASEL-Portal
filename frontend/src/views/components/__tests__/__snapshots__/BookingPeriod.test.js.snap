// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingPeriod renders properly 1`] = `
<span>
  Wed 06/10/21 06:46 PM  -  Sun 10/10/21 06:46 PM
</span>
`;
